source/ui/input-button-model.cpp

```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+
+#include "ui/input-button-model.hpp"
+#include "core/conversions.hpp"
+
+InputButtonModel::InputButtonModel(QQmlContext* context, MemoryAccess m) : QObject(), context(context), start(0), memoryAccess(m){
+    context->setContextProperty("inputButtonMod", this);
+}
+
+void InputButtonModel::buttonClicked(int id){
```

Comment:
You should also use an unsigned type for the id: `using id_t = unsigned long long` for example